Stix/Stix/Pages/Incident.cshtml.cs

```diff
@@ -21,5 +22,57 @@ public void OnGet()
             ViewData["IncidentAdded"] = dateTime;
             ViewData["IncidentSource"] = "https://www.definitely-a-valid-source.com/incident";
         }
+
+        [BindProperty]
+        public string FileFormatType { get; set; }
+
+        public IActionResult OnPost()
+        {
+            // {TO DO} Add try-catch blocks
+
+            // Store file format based on selected option
+            var format = Request.Form["file-formats"];
+
+            // Create spreadsheet
+            // Insert incident information in spreadsheet
+            // {TO DO} CHANGE THIS TO INSERT VARIABLES INSTEAD OF HARD CODED STRINGS
+            Workbook addIncidentInfoToExcelSpreadSheet()
+            {
+                Workbook workBook = new Workbook();
+
+                Worksheet workSheet = workBook.Worksheets[0];
+
+                Cell incidentNameCell = workSheet.Cells["A1"];
+                incidentNameCell.PutValue("Incident name");
+
+                Cell incidentDateCell = workSheet.Cells["B1"];
+                incidentDateCell.PutValue("Incident date " + DateTime.Now.ToString("MM/dd/yyyy HH:mm:ss", new CultureInfo("en-NL")));
+
+                Cell incidentAddedDateCell = workSheet.Cells["C1"];
+                incidentAddedDateCell.PutValue("Incident added date " + DateTime.Now.ToString("MM/dd/yyyy HH:mm:ss", new CultureInfo("en-NL")));
+
+                Cell incidentSourceCell = workSheet.Cells["D1"];
+                incidentSourceCell.PutValue("Incident source");
+
+                return workBook;
+            }
+
+            // Save filled in file in the selected format
+            void saveWorkBook(Workbook filledInFile)
+            {
+                filledInFile.Save("D:\\Incidents." + format);
```

Comment:
this throws exceptions and crashes if you dont have a D drive, should be changed to allow the user to select where to save